pkg/system/reconciler.go

```diff
@@ -342,6 +342,10 @@ func (r *Reconciler) Reconcile() (reconcile.Result, error) {
 			if err = r.deleteRootSecret(); err != nil {
 				log.Warnf("⏳ Temporary Error: %s", err)
 			}
+			if err = r.deleteOwnedBucketClasses(); err != nil {
```

Comment:
why do we need to do this `deleteOwnedBucketClasses` ?

Reply:
Kubernetes garbage collection doesn't work for cross namespace resources. This is to replicate that. Say, we have a bucketclass (created in namespace `b`) that is "owned" by noobaa-operator instance `1` (running in namespace `a`). Deletion of the system in this case will _not_ delete the bucketclass in the namespace `b` even if there is an owner reference.

Details: https://www.youtube.com/watch?v=iWz5AAbbT-c <- This is by Tom, he is at Red Hat.

Reply:
I don't think that the noobaa-operator label should mean ownership. The owner of the bucket-class is the entity that created it, and it should be responsible for deleting it. Even if it were in the same namespace, if the bucketclass owner is not noobaa CR then it would not be deleted by the garbage collector.